components/esp_lvgl_port/CMakeLists.txt

```diff
@@ -66,9 +76,22 @@ if("usb_host_hid" IN_LIST build_components)
     list(APPEND ADD_LIBS idf::usb_host_hid)
 endif()
 
-if(ADD_SRCS)
-    target_sources(${COMPONENT_LIB} PRIVATE ${ADD_SRCS})
-endif()
-if(ADD_LIBS)
-    target_link_libraries(${COMPONENT_LIB} PRIVATE ${ADD_LIBS})
-endif()
+# Here we create the real lvgl_port_lib
+add_library(lvgl_port_lib STATIC
+    ${PORT_PATH}/esp_lvgl_port.c
+    ${PORT_PATH}/esp_lvgl_port_disp.c
+    ${ADD_SRCS}
+    )
+target_include_directories(lvgl_port_lib PUBLIC "include")
+target_include_directories(lvgl_port_lib PRIVATE "priv_include")
+target_link_libraries(lvgl_port_lib PUBLIC
+    idf::esp_lcd
+    idf::${lvgl_name}
```

Comment:
I tested with `set(COMPONENTS main)` and did not encounter any problem.

I think the reason is that we require lvgl even from the idf_component.yml https://github.com/espressif/esp-bsp/blob/master/components/esp_lvgl_port/idf_component.yml#L6

So it can't be eliminated